pkg/ingress/controller/controller.go

```diff
@@ -312,6 +316,10 @@ func NewController(conf config.Config) *Controller {
 				// Two different versions of the same Ingress will always have different RVs.
 				return
 			}
+			newAnnotations := newIng.ObjectMeta.Annotations
+			oldAnnotations := oldIng.ObjectMeta.Annotations
+			delete(newAnnotations, "kubectl.kubernetes.io/last-applied-configuration")
+			delete(oldAnnotations, "kubectl.kubernetes.io/last-applied-configuration")
```

Comment:
what's the purpose for this? seems not related to the issue itself?

Reply:
Without this, a changement made in the annotations will not be applied.

Reply:
then I am curious why this is the first time we introduce this given we have so many annotations already

Reply:
from what I'v seen, the `UpdateEvent` was not triggered when an annotation were updated. Hence these lines, and the condition line 333.

```golang
!reflect.DeepEqual(newAnnotations, oldAnnotations)
```

Reply:
ok, thanks for the info~